packages/alchemy/src/config/store/core.ts

```diff
@@ -1,30 +1,116 @@
+import { DEFAULT_STORAGE_KEY } from "./client.js";
+import type { CoreState, CoreStore } from "./types.js";
 
 export type CreateCoreStoreParams = {
-  connection: ConnectionConfig;
+  connections: Connection[];
   chain: Chain;
+  storage?: Storage;
+  ssr?: boolean;
 };
 
-export const createCoreStore = ({
-  connection,
-  chain,
-}: CreateCoreStoreParams) => {
-  const bundlerClient = createAlchemyPublicRpcClient({
+/**
+ * Create the core store for alchemy accounts. This store contains the bundler client
+ * as well as the chain configs (including the initial chain to use)
+ *
+ * @param params connections configs
+ * @param params.connections a collection of chains and their connection configs
+ * @param params.chain the initial chain to use
+ * @param params.storage the storage to use for persisting the state
+ * @param params.ssr whether the store is being created on the server
+ * @returns the core store
+ */
+export const createCoreStore = (params: CreateCoreStoreParams): CoreStore => {
+  const {
+    connections,
     chain,
-    connectionConfig: connection,
-  });
+    storage = typeof window !== "undefined" ? localStorage : undefined,
+    ssr,
+  } = params;
 
   // State defined in here should work either on the server or on the client
   // bundler client for example can be used in either setting to make RPC calls
   const coreStore = createStore(
-    subscribeWithSelector(() => ({
-      bundlerClient,
-    }))
+    subscribeWithSelector(
+      storage
+        ? persist(() => createInitialCoreState(connections, chain), {
+            name: `${DEFAULT_STORAGE_KEY}:core`,
+            storage: createJSONStorage<CoreState>(() => storage, {
+              replacer: (key, value) => {
+                if (key === "bundlerClient") {
+                  const client = value as CoreState["bundlerClient"];
+                  return {
+                    connection: connections.find(
+                      (x) => x.chain.id === client.chain.id
+                    ),
+                  };
+                }
+                return bigintMapReplacer(key, value);
+              },
+              reviver: (key, value) => {
+                if (key === "bundlerClient") {
+                  const connection = value as Connection;
+                  return createAlchemyPublicRpcClient({
+                    chain: connection.chain,
+                    connectionConfig: connection,
+                  });
+                }
+
+                return bigintMapReviver(key, value);
+              },
+            }),
+            skipHydration: ssr,
+          })
+        : () => createInitialCoreState(connections, chain)
+    )
+  );
+
+  coreStore.subscribe(
+    ({ chain }) => chain,
+    (chain) => {
+      coreStore.setState({
+        bundlerClient: createAlchemyPublicRpcClient({
+          chain,
+          connectionConfig: coreStore.getState().connections.get(chain.id)!,
+        }),
+      });
+    }
```

Comment:
also gonna do this as part of `setChain` so that these things get set at the same time